aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1337,20 +1334,118 @@ Tensor& cholesky_out(const Tensor &self, bool upper, Tensor &result) {
   return result;
 }
 
-Tensor linalg_cholesky(const Tensor &self) {
-  if (self.numel() == 0) {
-    return at::empty_like(self, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+DEFINE_DISPATCH(cholesky_stub);
+
+void linalg_cholesky_out_info(const Tensor& input, const Tensor& result, const Tensor& info) {
+  TORCH_INTERNAL_ASSERT(input.dim() >= 2);
+  TORCH_INTERNAL_ASSERT(input.size(-1) == input.size(-2));
+
+  TORCH_INTERNAL_ASSERT(result.scalar_type() == input.scalar_type());
+  TORCH_INTERNAL_ASSERT(result.device() == input.device());
+
+  TORCH_INTERNAL_ASSERT(info.scalar_type() == at::kInt);
+  TORCH_INTERNAL_ASSERT(info.device() == input.device());
+
+  // if result has no elements we can modify it
+  if (result.numel() == 0) {
+    at::native::resize_as_(result, input.transpose(-2, -1), MemoryFormat::Contiguous);
+    result.transpose_(-2, -1);
   }
-  squareCheckInputs(self);
-  return at::_cholesky_helper(self, /*upper=*/false).tril_();
+
+  // result tensor must be in batched column major order (Fortran contiguous)
+  TORCH_INTERNAL_ASSERT(result.transpose(-2, -1).is_contiguous());
+  TORCH_INTERNAL_ASSERT(result.sizes().equals(input.sizes()));
+
+  // cholesky_stub (apply_cholesky) performs calculations in-place and result must be a copy of input
+  result.copy_(input);
+
+  // if info has no elements we can modify it
+  auto expected_info_shape = IntArrayRef(input.sizes().cbegin(), input.sizes().cend() - 2); // input.shape[:-2]
+  if (info.numel() == 0) {
+    info.resize_(expected_info_shape);
+  }
+
+  // info must be contiguous
+  TORCH_INTERNAL_ASSERT(info.is_contiguous());
+  TORCH_INTERNAL_ASSERT(info.sizes().equals(expected_info_shape));
+  info.fill_(0);
+
+  cholesky_stub(result.device().type(), result, info);
+
+  result.tril_();
+}
+
+std::tuple<Tensor&, Tensor&> linalg_cholesky_ex_out(const Tensor& input, bool check_errors, Tensor& L, Tensor& info) {
+  squareCheckInputs(input);
+  checkSameDevice("torch.linalg.cholesky_ex", L, input, "L");
```

Comment:
Because cholesky is implemented as a call to cholesky_ex this would show the name cholesky_ex if a call to cholesky hit an error, right? Also seem comment below for when a call to cholesky_ex would show the name cholesky

Error messages aren't the end of the world but do you think we should bother taking a string for the function name? Alternatively we could just always use torch.linalg.cholesky, even when someone calls torch.linalg.cholesky_ex

Reply:
I agree that we could just use torch.linalg.cholesky.
Here `torch.linalg.cholesky` has its own same checks differing in the function name and that "result" is used instead of "L". It duplicates the check, but that shouldn't have overhead.

Reply:
So the error messages of `torch.linalg.cholesky` are not changed in this PR. The tests would need to be modified if the error messages would change.